wire-ios-sync-engine/Source/UserSession/URLActionProcessors/DeepLinkURLActionProcessor.swift

```diff
@@ -36,96 +36,129 @@ class DeepLinkURLActionProcessor: URLActionProcessor {
-                    // First of all, we should try to fetch the conversation with ID from the response.
-                    // If the conversation doesn't exist, we should initiate a request to join the conversation
-                    if let conversation = ZMConversation.fetch(with: conversationId, in: viewContext),
-                       conversation.isSelfAnActiveMember {
-                        delegate.showConversation(conversation, at: nil)
-                        delegate.completedURLAction(urlAction)
-                    } else {
-                        delegate.shouldPerformActionWithMessage(conversationName, action: urlAction) { shouldJoin in
-
-                            guard shouldJoin else {
-                                delegate.completedURLAction(urlAction)
-                                return
-                            }
-
-                            ZMConversation.join(
-                                key: key,
-                                code: code,
-                                transportSession: strongSelf.transportSession,
-                                eventProcessor: strongSelf.eventProcessor,
-                                contextProvider: strongSelf.contextProvider
-                            ) { [weak self] response in
-
-                                guard let strongSelf = self else { return }
-
-                                switch response {
-                                case .success(let conversation):
-                                    strongSelf.synchronise(conversation) { result in
-                                        DispatchQueue.main.async {
-                                            switch result {
-                                            case .success(let syncConversation):
-                                                delegate.showConversation(syncConversation, at: nil)
-                                            case .failure(let error):
-                                                delegate.failedToPerformAction(urlAction, error: error)
-                                            }
-                                        }
-                                    }
-                                case .failure(let error):
-                                    delegate.failedToPerformAction(urlAction, error: error)
-                                }
-
-                                delegate.completedURLAction(urlAction)
-                            }
-                        }
-                    }
+        case .openConversation(let id):
+            handleOpenConversation(id: id, delegate: delegate)
+
+        case .openUserProfile(let id):
+            handleOpenUserProfile(id: id, delegate: delegate)
+
+        default:
+            delegate?.completedURLAction(urlAction)
+        }
+    }
+
+    private func handleJoinConversation(key: String, code: String, urlAction: URLAction, delegate: PresentationDelegate?) {
```

Comment:
please consider line breaks for parameters :)

Reply:
Done! https://github.com/wireapp/wire-ios/commit/b5637f5bab11bb4951fa918b66878443cc2ad479